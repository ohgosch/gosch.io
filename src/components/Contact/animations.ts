import { keyframes } from 'styled-components';

export const LoaderSectionAnimation = keyframes`
  from {
    opacity: 0;
  }
  
  to {
    opacity: 1;
  }
`;

export const LoaderIconAnimation = keyframes`
  from {
    transform: rotate(0deg);
  }
  
  to {
    transform: rotate(360deg);
  }
`;
