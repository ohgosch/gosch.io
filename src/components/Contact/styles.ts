import styled, { css } from 'styled-components';
import { rem } from 'polished';

import { desktop, mobile, mobileTablet, tabletDesktop } from 'visual/medias';

export const Container = styled.section``;

export const Content = styled.form`
  ${desktop(css`
    width: ${rem(528)};
    margin: 0 auto;
    display: flex;
    flex-wrap: wrap;
    justify-content: flex-end;
  `)}
`;

export const InputContent = styled.div`
  width: 100%;
  display: grid;
  justify-content: center;
  grid-gap: ${rem(16)};
  margin-bottom: ${rem(24)};

  ${tabletDesktop(css`
    grid-template-columns: repeat(2, 1fr);
  `)}

  ${desktop(css`
    margin-top: ${rem(64)};
  `)}

    ${mobileTablet(css`
    margin-top: ${rem(48)};
  `)}
    
    ${mobile(css`
    grid-template-columns: 1fr;
  `)}
`;
