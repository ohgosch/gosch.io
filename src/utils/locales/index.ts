import I18nextBrowserLanguageDetector from 'i18next-browser-languagedetector';
import I18NextLocizeBackend from 'i18next-locize-backend';
import { initReactI18next } from 'react-i18next';
import i18n from 'i18next';

import ptBR from './ptBR.json';
import en from './en.json';

i18n
  .use(I18NextLocizeBackend)
  .use(initReactI18next)
  .use(I18nextBrowserLanguageDetector)
  .init({
    debug: false,
    fallbackLng: 'en',
    whitelist: ['pt', 'en'],
    supportedLngs: ['pt', 'en'],
    detection: {
      order: ['querystring', 'navigator'],
      lookupQuerystring: 'lng',
    },
    resources: {
      en: {
        translation: en,
      },
      pt: {
        translation: ptBR,
      },
    },
  });

export default i18n;
