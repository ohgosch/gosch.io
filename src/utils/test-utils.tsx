import { ThemeProvider } from 'styled-components';
import { render } from '@testing-library/react';
import { ReactElement } from 'react';

import theme from 'visual/theme';

const customRender = (ui: ReactElement) =>
  render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>);

export * from '@testing-library/react';
export { customRender as render };
