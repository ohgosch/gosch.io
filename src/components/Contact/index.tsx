import Input from 'visual/styles/Input';

import * as S from './styles';
import Heading1 from 'visual/styles/Heading1';
import Wrapper from 'visual/styles/Wrapper';
import Button from 'visual/styles/Button';

const Contact = () => (
  <Wrapper id="contact">
    <S.Container>
      <Heading1>Contact</Heading1>
      <S.Content>
        <S.InputContent>
          <Input required name="name" placeholder="Name" />
          <Input required name="email" placeholder="E-mail" type="email" />
          <Input required name="company" placeholder="Company" />
          <Input required name="role" placeholder="Role" />
        </S.InputContent>
        <Button>Submit</Button>
      </S.Content>
    </S.Container>
  </Wrapper>
);

export default Contact;
