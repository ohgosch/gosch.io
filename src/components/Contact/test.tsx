import { render, screen } from 'utils/test-utils';

import Contact from '.';

describe('<Contact />', () => {
  it('should render correctly', () => {
    const { container } = render(<Contact />);

    expect(
      screen.getByRole('heading', { name: /Contact/i }),
    ).toBeInTheDocument();

    expect(screen.getByPlaceholderText(/Name/i)).toBeInTheDocument();
    expect(screen.getByPlaceholderText(/E-mail/i)).toBeInTheDocument();
    expect(screen.getByPlaceholderText(/Company/i)).toBeInTheDocument();
    expect(screen.getByPlaceholderText(/Role/i)).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /Submit/i }));

    expect(container.firstChild).toMatchSnapshot();
  });
});
