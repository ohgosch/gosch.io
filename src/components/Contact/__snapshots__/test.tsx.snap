// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Contact /> should render correctly 1`] = `
.c4 {
  height: 3.5rem;
  padding: 0 1rem;
  background-color: transparent;
  border-radius: 0.25rem;
  border: 0.125rem solid #DBA14B;
  color: #F2F2F2;
  font-size: 1.25rem;
}

.c4::-webkit-input-placeholder {
  color: rgba(242,242,242,0.7);
}

.c4::-moz-placeholder {
  color: rgba(242,242,242,0.7);
}

.c4:-ms-input-placeholder {
  color: rgba(242,242,242,0.7);
}

.c4::placeholder {
  color: rgba(242,242,242,0.7);
}

.c3 {
  width: 100%;
  display: grid;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  grid-gap: 1rem;
  margin-bottom: 1.5rem;
}

.c1 {
  font-weight: bold;
  font-size: 3rem;
  line-height: 3.5rem;
  text-align: center;
  -webkit-letter-spacing: 0.08em;
  -moz-letter-spacing: 0.08em;
  -ms-letter-spacing: 0.08em;
  letter-spacing: 0.08em;
  text-transform: uppercase;
  cursor: default;
}

.c0 {
  margin-left: auto;
  margin-right: auto;
}

.c5 {
  height: 2.5rem;
  padding: 0 1.5rem;
  font-style: normal;
  font-weight: bold;
  font-size: 0.875rem;
  color: #F2F2F2;
  background-color: #DBA14B;
  border-radius: 0.25rem;
}

@media (min-width:64.0625rem) {
  .c2 {
    width: 33rem;
    margin: 0 auto;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    -webkit-box-pack: end;
    -webkit-justify-content: flex-end;
    -ms-flex-pack: end;
    justify-content: flex-end;
  }
}

@media (min-width:46.0625rem) {
  .c3 {
    grid-template-columns: repeat(2,1fr);
  }
}

@media (min-width:64.0625rem) {
  .c3 {
    margin-top: 4rem;
  }
}

@media (max-width:64rem) {
  .c3 {
    margin-top: 3rem;
  }
}

@media (max-width:46rem) {
  .c3 {
    grid-template-columns: 1fr;
  }
}

@media (max-width:46rem) {
  .c0 {
    width: 90%;
    max-width: 46rem;
  }
}

@media (min-width:46.0625rem) and (max-width:64rem) {
  .c0 {
    width: 88%;
    max-width: 64rem;
  }
}

@media (min-width:64.0625rem) {
  .c0 {
    width: 90%;
    max-width: 64.0625rem;
  }
}

<div
  class="c0"
  id="contact"
>
  <section
    class=""
  >
    <h2
      class="c1"
    >
      Contact
    </h2>
    <form
      class="c2"
    >
      <div
        class="c3"
      >
        <input
          class="c4"
          name="name"
          placeholder="Name"
          required=""
        />
        <input
          class="c4"
          name="email"
          placeholder="E-mail"
          required=""
          type="email"
        />
        <input
          class="c4"
          name="company"
          placeholder="Company"
          required=""
        />
        <input
          class="c4"
          name="role"
          placeholder="Role"
          required=""
        />
      </div>
      <button
        class="c5"
      >
        Submit
      </button>
    </form>
  </section>
</div>
`;
